com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs

```diff
@@ -95,16 +95,27 @@ private void GetTransparencySortingMode(Camera camera, ref SortingSettings sorti
 
                     using(new ProfilingScope(cmd, m_ProfilingDrawLayerBatch))
                     {
-                        unsafe
+                        if (layerBatch.lightStats.totalLights > 0)
                         {
-                            for (var blendStyleIndex = 0; blendStyleIndex < k_ShapeLightTextureIDs.Length; blendStyleIndex++)
+                            unsafe
                             {
-                                var used = layerBatch.renderTargetUsed[blendStyleIndex];
+                                for (var blendStyleIndex = 0; blendStyleIndex < k_ShapeLightTextureIDs.Length; blendStyleIndex++)
+                                {
+                                    var used = layerBatch.renderTargetUsed[blendStyleIndex];
 
-                                if (used)
-                                    cmd.SetGlobalTexture(k_ShapeLightTextureIDs[blendStyleIndex], new RenderTargetIdentifier(layerBatch.renderTargetIds[blendStyleIndex]));
+                                    if (used)
+                                        cmd.SetGlobalTexture(k_ShapeLightTextureIDs[blendStyleIndex], new RenderTargetIdentifier(layerBatch.renderTargetIds[blendStyleIndex]));
 
-                                RendererLighting.EnableBlendStyle(cmd, blendStyleIndex, used);
+                                    RendererLighting.EnableBlendStyle(cmd, blendStyleIndex, used);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            for (var blendStyleIndex = 0; blendStyleIndex < k_ShapeLightTextureIDs.Length; blendStyleIndex++)
+                            {
+                                cmd.SetGlobalTexture(k_ShapeLightTextureIDs[blendStyleIndex], Texture2D.blackTexture);
+                                RendererLighting.EnableBlendStyle(cmd, blendStyleIndex, blendStyleIndex == 0 ? true : false);
```

Comment:
this should be set to `false` for all except maybe just one?

Reply:
ah, the tertiary is already doing that. But it could be simplified...

Reply:
Fixed.